sonic-xcvrd/scripts/xcvrd

```diff
@@ -33,15 +33,26 @@ PLATFORM_SPECIFIC_CLASS_NAME = "SfpUtil"
 
 TRANSCEIVER_INFO_TABLE = 'TRANSCEIVER_INFO'
 TRANSCEIVER_DOM_SENSOR_TABLE = 'TRANSCEIVER_DOM_SENSOR'
+TRANSCEIVER_ERROR_TABLE = 'TRANSCEIVER_ERROR'
 
 SELECT_TIMEOUT_MSECS = 1000
 
 DOM_INFO_UPDATE_PERIOD_SECS = 60
 TIME_FOR_SFP_READY_SECS = 1
 XCVRD_MAIN_THREAD_SLEEP_SECS = 60
 
-SFP_STATUS_INSERTED = '1'
+# SFP status definition, shall be aligned with the definition in get_change_event() of ChassisBase
 SFP_STATUS_REMOVED = '0'
+SFP_STATUS_INSERTED = '1'
+SFP_STATUS_ERR_I2C_STUCK = '2'
+SFP_STATUS_ERR_BAD_EEPROM = '3'
+SFP_STATUS_ERR_UNSUPPORTED_CABLE = '4'
+SFP_STATUS_ERR_HIGH_TEMP = '5'
+SFP_STATUS_ERR_BAD_CABLE = '6'
+
+error_block_eeprom_list = [SFP_STATUS_ERR_I2C_STUCK, SFP_STATUS_ERR_BAD_EEPROM,
+                           SFP_STATUS_ERR_UNSUPPORTED_CABLE, SFP_STATUS_ERR_HIGH_TEMP,
+                           SFP_STATUS_ERR_BAD_CABLE]
```

Comment:
Could you use a Enum instead of list?

Reply:
revised, but the way package enum34 is not installed in pmon, I add a commit in PR https://github.com/Azure/sonic-buildimage/pull/4365 to install it.